flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/deployment/AbstractDeploymentObserver.java

```diff
@@ -141,41 +148,19 @@ protected void observeJmDeployment(
             return;
         }
 
-        Optional<Deployment> deployment = context.getSecondaryResource(Deployment.class);
-        if (deployment.isPresent()) {
-            DeploymentStatus status = deployment.get().getStatus();
-            DeploymentSpec spec = deployment.get().getSpec();
-            if (status != null
-                    && status.getAvailableReplicas() != null
-                    && spec.getReplicas().intValue() == status.getReplicas()
-                    && spec.getReplicas().intValue() == status.getAvailableReplicas()
-                    && flinkService.isJobManagerPortReady(effectiveConfig)) {
-
-                // typically it takes a few seconds for the REST server to be ready
-                logger.info(
-                        "JobManager deployment port is ready, waiting for the Flink REST API...");
-                deploymentStatus.setJobManagerDeploymentStatus(
-                        JobManagerDeploymentStatus.DEPLOYED_NOT_READY);
+        if (KubernetesDeploymentMode.NATIVE.equals(flinkApp.getSpec().getMode())) {
+            Optional<Deployment> deployment = context.getSecondaryResource(Deployment.class);
+            if (deployment.isPresent()) {
+                jmDeploymentObserver(deployment.get(), effectiveConfig, deploymentStatus, flinkApp);
                 return;
```

Comment:
Why do we use StatefulSets for JobManager also? Seems to be an unnecessary complication without much benefits.

Reply:
With StatefulSet for JobManager has the following advantages：

1. Avoid random string suffix of JM Pod, more intuitive identification of JM.
2. There is no need to manually create the PVC each time, which is very convenient when you need to create many flink clusters in production.
3. When JM replicas is greater than 1, the one-to-one correspondence between PVC and JM can be guaranteed.

Reply:
If these are the only advantages I would prefer Deployment:
 1. Native integration has the random suffix, not a problem you can select by label if necessary
 2. Not sure why you need a PVC for the JM
 3. There is not a good reason to have more than one 1 replica and even then PVC is not necessary anyways
 
 @wangyang0918

Reply:
Also looping @tweise here

Reply:
Another benefit:
when using a system with StatefulSet, allowing a simple HA solution like using a filesystem with PVC(such as SSD) can recover faster than remote storage(such as HDFS or OSS).

Reply:
It's very important to run JM with PVC for working with work-dir[1] to fast recovery.

1. https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/deployment/resource-providers/standalone/working_directory/

Reply:
I think you can have PVC with the current podTemplate also right?

Also the JM doesn’t access much data during recovery , so even if it has to fetch it from remote storage it wouldn’t really have a big impact.

Reply:
Sorry for join the discussion so late.

I share the same concern with @gyfora and @usamj . Having a PVC for JobManager does not take enough benefits now. IIRC, we does not have any state in the JobManager which needs to be recovered after failover. Using the StatefulSet also makes the JobManager has risk that could not be started if the previous one terminated too slow or stucked.